service/history/handler.go

```diff
@@ -993,6 +993,42 @@ func (h *Handler) ResetWorkflowExecution(ctx context.Context,
 	return resp, nil
 }
 
+// QueryWorkflow queries a workflow.
+func (h *Handler) QueryWorkflow(
+	ctx context.Context,
+	request *hist.QueryWorkflowRequest,
+) (resp *hist.QueryWorkflowResponse, retError error) {
+	defer log.CapturePanic(h.GetLogger(), &retError)
+	h.startWG.Wait()
+
+	scope := metrics.HistoryQueryWorkflowScope
+	h.metricsClient.IncCounter(scope, metrics.CadenceRequests)
+	sw := h.metricsClient.StartTimer(scope, metrics.CadenceLatency)
+	defer sw.Stop()
+
+	domainID := request.GetDomainUUID()
+	if domainID == "" {
+		return nil, h.error(errDomainNotSet, scope, domainID, "")
+	}
+
+	if ok := h.rateLimiter.Allow(); !ok {
+		return nil, h.error(errHistoryHostThrottle, scope, domainID, "")
+	}
+
+	workflowID := request.GetExecution().GetWorkflowId()
+	engine, err1 := h.controller.GetEngine(workflowID)
```

Comment:
these can just be `err` do not need `err1` and `err2`

Reply:
Yes I agree but we use this pattern throughout all other methods in this file.

Reply:
If something is an unnecessary pattern we should change it - so over time the codebase gets cleaned up. I normally like changing the file I am working on - so at least future uses will not use the pattern.. 

In general using 2 error variables can get misleading, unless necessary - especially if they are not scoped like the below. You have 2 valid error variable and people may miss what to return and work with.. 

```
if customErr := op1; customErr != nil {
```